sql/core.go

```diff
@@ -421,6 +423,118 @@ type Histogram struct {
 // HistogramMap is a map from column name to associated histogram
 type HistogramMap map[string]*Histogram
 
+// HistogramMapBuilder will construct a HistogramMap given a PartitionIterator and Schema
+// TODO: have option for number of buckets (and logic to convert freqMap into those buckets)
+// TODO: could iterate over Partitions asynchronously (after exchange is rewritten)
+func HistogramMapBuilder(ctx *Context, t Table) (HistogramMap, error) {
```

Comment:
this is written as a constructor, which we use `New...` to indicate. A builder is a class that exposes an interface for incrementally constructing the object, which when finished is returned with `builder.Build`. If you write it as a constructor, i would recommend taking `sql.RowIter` as an argument, not a table.